header-rewrite-filter/header_rewrite_integration_test.cc

```diff
@@ -41,8 +50,8 @@ TEST_P(HttpFilterHeaderRewriteIntegrationTest, Test1) {
   ASSERT_TRUE(response->waitForEndStream());
 
   EXPECT_EQ(
-      "sample-filter",
-      request_stream->headers().get(Http::LowerCaseString("via"))[0]->value().getStringView());
+      "http,https",
```

Comment:
I think this should just be https to match L34 - is there a reason the value is appended instead of set?

Reply:
I mentioned this in a previous PR, but the current implementation of set-header appends to the header rather than replacing what was there. This is to cover the case that multiple values are needing to be set. Is the desired behavior to first replace what was previously in the header and then append to it if there are additional values to set?

Reply:
Both would be useful.
Maybe 2 directives, e.g. `set-header` and `replace-header`?

Reply:
@henrymwang Will add this to the backlog if you agree!

Reply:
To me, set-header and replace-header mean the same thing. Maybe we can do set-header and append-header instead?

Reply:
I don't feel strongly about the names. 
But incidentally, both HAProxy and ATS use `set-header` and `add-header`
http://docs.haproxy.org/2.4/configuration.html#http-request%20replace-header
https://docs.trafficserver.apache.org/en/7.1.x/admin-guide/plugins/header_rewrite.en.html